packages/block-editor/src/components/block-parent-selector/index.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * WordPress dependencies
+ */
+import { getBlockType } from '@wordpress/blocks';
+import { Button } from '@wordpress/components';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import BlockIcon from '../block-icon';
+
+/**
+ * Block parent selector component, displaying the hierarchy of the
+ * current block selection as a single icon to "go up" a level.
+ *
+ * @return {WPElement} Parent block selector.
+ */
+export default function BlockParentSelector() {
+	const { selectBlock } = useDispatch( 'core/block-editor' );
+	const { parents } = useSelect( ( select ) => {
```

Comment:
Seems  like you only  need a boolean "hasParents". When using useSelect,  it's better to limit the result to things that change less often. I assume "parents" here is a block list, it changes way too often compared to a boolean (that's a performance trick)

Reply:
`parents` is being used to calculate `firstParentClientId`, so it is needed. However, if we combine the `useSelect` calls, we don't have to return `parents` directly; we could calculate `hasParents` in the select callback and return it, and thus simplify `if ( parents?.length )` to `if ( hasParents )`.

Reply:
I removed `parent` from the `useSelect` return, and instead used the `firstParentClientId` to do the check. Maybe this is a terrible idea?

Reply:
Is `firstParentClientId` a string or a number? If it's a number, is it possible for `firstParentClientId` to be something like `0`, which would be coerced to `false`? I think for the sake of clarity and avoiding issues like that, it may be best to just use a `hasParents` boolean.

Alternatively, you could explicitly check if it's `undefined`:
```js
if ( firstParentClientId !== undefined ) {
```

Reply:
I added the explicit `undefined` check.